TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/EditName/Confirm.cshtml.cs

```diff
@@ -0,0 +1,77 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeachingRecordSystem.Core.Dqt.Models;
+using TeachingRecordSystem.Core.Dqt.Queries;
+
+namespace TeachingRecordSystem.SupportUi.Pages.Persons.PersonDetail.EditName;
+
+[Journey(JourneyNames.EditName), RequireJourneyInstance]
+public class ConfirmModel : PageModel
+{
+    private readonly TrsLinkGenerator _linkGenerator;
+    private readonly ICrmQueryDispatcher _crmQueryDispatcher;
+
+    public ConfirmModel(
+        TrsLinkGenerator linkGenerator,
+        ICrmQueryDispatcher crmQueryDispatcher)
+    {
+        _linkGenerator = linkGenerator;
+        _crmQueryDispatcher = crmQueryDispatcher;
+    }
+
+    public JourneyInstance<EditNameState>? JourneyInstance { get; set; }
+
+    [FromRoute]
+    public Guid PersonId { get; set; }
+
+    public string? CurrentValue { get; set; }
+
+    public string? NewValue { get; set; }
+
+    public async Task<IActionResult> OnPost()
+    {
+        await _crmQueryDispatcher.ExecuteQuery(
+            new UpdateContactNameQuery(
+                PersonId,
+                JourneyInstance!.State.FirstName,
+                JourneyInstance!.State.MiddleName,
+                JourneyInstance!.State.LastName));
+
+        await JourneyInstance!.CompleteAsync();
+
+        TempData.SetFlashSuccess("Record has been updated");
+
+        return Redirect(_linkGenerator.PersonDetail(PersonId));
+    }
+
+    public override async Task OnPageHandlerExecutionAsync(PageHandlerExecutingContext context, PageHandlerExecutionDelegate next)
+    {
+        var person = await _crmQueryDispatcher.ExecuteQuery(
+            new GetContactDetailByIdQuery(
+                PersonId,
+                new ColumnSet(
+                    Contact.PrimaryIdAttribute,
+                    Contact.Fields.FirstName,
+                    Contact.Fields.MiddleName,
+                    Contact.Fields.LastName)));
+
+        if (!JourneyInstance!.State.IsComplete)
```

Comment:
Nit: we could move this check above the CRM query (as it's cheaper)

Reply:
Done